tests/unit/core/test-class-mode.php

```diff
@@ -57,6 +57,31 @@ public function test_init_enters_dev_mode_when_constant_is_defined() {
 		$this->assertFalse( $this->mode->is_live() );
 	}
 
+	public function provider_init_enters_dev_mode_for_WP_DEVELOPMENT_MODE(): array {
+		return [
+			'core'         => [ 'core', true ],
+			'plugin'       => [ 'plugin', true ],
+			'theme'        => [ 'theme', true ],
+			'all'          => [ 'all', true ],
+			'empty string' => [ '', false ],
+		];
+	}
+
+	/**
+	 * @dataProvider provider_init_enters_dev_mode_for_WP_DEVELOPMENT_MODE
+	 */
+	public function test_init_enters_dev_mode_for_WP_DEVELOPMENT_MODE( ?string $constant_value, bool $is_dev ) {
```

Comment:
```suggestion
	public function test_init_enters_dev_mode_for_WP_DEVELOPMENT_MODE( ?string $constant_value, bool $should_init_in_dev_mode ) {
```

Just for readability

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/10275/commits/67f1ac75cc146adb739cf0bc9b87680ed450234e